actions/utils.ts

```diff
@@ -85,15 +156,114 @@ function handlePromiseErrors<T>(
     .then(() => true)
     .catch((error) => {
       console.error(errorMessage, error);
-      return true;
+      return false;
     });
 }
 
 /**
- * Convenient utility to log in case theres an error writing in the registry
+ * Convenient utility to log in case theres an error writing in the registry and return a boolean with the result of the operation
+ *
  * @param registry Tenderly registry
  * @returns a promise that returns true if the registry write was successful
  */
-export function writeRegistry(registry: Registry): Promise<boolean> {
-  return handlePromiseErrors("Error writing registry", registry.write());
+export async function writeRegistry(): Promise<boolean> {
+  if (executionContext) {
+    return handlePromiseErrors(
+      "Error writing registry",
+      executionContext.registry.write()
+    );
+  }
+
+  return true;
+}
+
+var consoleOriginal = {
+  log: console.log,
+  error: console.error,
+  warn: console.warn,
+  debug: console.debug,
+};
+
+/**
+ * Tenderly has a limit of 4Kb per log message. When you surpas this limit, the log is not printed any more making it super hard to debug anythnig
+ *
+ * This tool will print
+ *
+ * @param data T
+ */
+const logWithLimit =
+  (level: "log" | "warn" | "error" | "debug") =>
+  (...data: any[]) => {
+    const bigLogText = data
+      .map((item) => {
+        if (typeof item === "string") {
+          return item;
+        }
+        return JSON.stringify(item, null, 2);
+      })
+      .join(" ");
+
+    const numChunks = Math.ceil(bigLogText.length / TENDERLY_LOG_LIMIT);
+
+    for (let i = 0; i < numChunks; i += 1) {
+      const chartStart = i * TENDERLY_LOG_LIMIT;
+      const prefix = numChunks > 1 ? `[${i + 1}/${numChunks}] ` : "";
+      const message =
+        prefix +
+        bigLogText.substring(chartStart, chartStart + TENDERLY_LOG_LIMIT);
+      consoleOriginal[level](message);
+
+      // if (level === "error") {
+      //   sendSlack(message);
+      // }
+
+      // // Used to debug the Tender log Limit issues
+      // consoleOriginal[level](
+      //   prefix + "TEST for bigLogText of " + bigLogText.length + " bytes"
+      // );
```

Comment:
I can remove if it bothers the reviewers. It help me a lot in debugging the issues with the logs being removed. I don't consider the problem 100% solved, so i left it as a comment to quickly uncomment if needed